src/agent/onefuzz-task/src/tasks/coverage/generic.rs

```diff
@@ -138,9 +138,34 @@ impl CoverageTask {
             return Ok(CmdFilter::default());
         };
 
+        let setup_dir = &self.config.common.setup_dir;
+
         // Ensure users can locate the filter relative to the setup container.
-        let expand = Expand::new().setup_dir(&self.config.common.setup_dir);
-        let filter_path = expand.evaluate_value(raw_filter_path)?;
+        let expand = Expand::new().setup_dir(setup_dir);
+        let expanded_filter_path = expand.evaluate_value(raw_filter_path)?;
+
+        // Determine if `coverage_filter` is already prefixed with a absolute or relative
```

Comment:
Hmm, we can do this more robustly by introducing a `find_setup_relative_file()` function which could use file existence checks and attempted expansions to handle the cases of no prefixing (a fully relativized subpath), `setup/`-prefixing, and `{setup_dir}`-prefixing. We could use that same function wherever we are prepending `{setup_dir}` to `target_exe`. Let's try that instead.